lib/cloudfront-stack.ts

```diff
@@ -1,20 +1,48 @@
-import * as cloudfront from 'aws-cdk-lib/aws-cloudfront';
-import * as origins from 'aws-cdk-lib/aws-cloudfront-origins';
 import { Construct } from 'constructs';
+import * as cloudfront from 'aws-cdk-lib/aws-cloudfront';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { PriceClass, ViewerProtocolPolicy } from 'aws-cdk-lib/aws-cloudfront';
 import { S3Stack } from './s3-stack';
+import { AcmStack } from './acm-stack';
+import getEnv from '../shared/getEnv';
+import { Bucket } from 'aws-cdk-lib/aws-s3';
+import { S3OriginWithOACPatch } from './s3-origin-with-oac-patch';
 
 export class CloudfrontStack {
   public readonly distribution: cloudfront.Distribution;
 
-  constructor(scope: Construct, s3: S3Stack) {
-    const origin = new origins.S3Origin(s3.bucket);
-    this.distribution = new cloudfront.Distribution(scope, 'cdk-cloudfront', {
+  constructor(scope: Construct, s3: S3Stack, acm: AcmStack) {
+    const s3BucketOAC = new cloudfront.CfnOriginAccessControl(scope, 'cdk-s3-bucket-OAC', {
+      originAccessControlConfig: {
+        name: 's3-bucket-OAC',
+        originAccessControlOriginType: 's3',
+        signingBehavior: 'always',
+        signingProtocol: 'sigv4',
+      },
+    });
+    const s3BucketOrigin = new S3OriginWithOACPatch(
+      Bucket.fromBucketName(scope, 'cdk-bucket-copy', s3.bucket.bucketName),
+      { oacId: s3BucketOAC.getAtt('Id') }
+    );
+    const domainNames = getEnv('DOMAIN_NAME_LIST')?.split(',');
+    if (!domainNames) return;
+    this.distribution = new cloudfront.Distribution(scope, 'cdk-cloudfront-distribution', {
```

Comment:
phần đặt tên này đang bị lỗi validate domainName nha 
 1 validation error detected: Value 'cdk-cloudfront-distribution' at 'domainName' failed to satisfy constraint: Member must satisfy 
regular expression pattern: ^(\*\.)?(((?!-)[A-Za-z0-9-]{0,62}[A-Za-z0-9])\.)+((?!-)[A-Za-z0-9-]{1,62}[A-Za-z0-9])